### PR TITLE
openthread: fixes for supported Thread 1.2 features

### PR DESCRIPTION
**Pull request by edmont**
Disable retransmissions when using radio transmission security capabilities and adapt Matter and docs to the change.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

KRKNWK-10467

### CODE REVIEW

**Review by tejlmand — APPROVED** [review]

**Review by czeslawmakarski — APPROVED** [review]

File: doc/nrf/ug_thread_configuring.rst [hunk 1]
```diff
@@ -129,9 +129,10 @@ You can change the stack version by using the following Kconfig options:
 
 By selecting support for Thread 1.2, you enable the following features in addition to the :ref:`Thread 1.1 features <thread_ug_supported_features>`:
 
-* Enhanced Frame Pending
-* Enhanced Keep Alive
-* Thread Domain Name
+* Enhanced Frame Pending.
+* Enhanced Keep Alive.
+* Thread Domain Name.
```
Comment (greg-fer) [inline doc/nrf/ug_thread_configuring.rst:134]: No full periods if the entry is not a complete sentence.

File: doc/nrf/ug_thread_configuring.rst [hunk 2]
```diff
@@ -129,9 +129,10 @@ You can change the stack version by using the following Kconfig options:
...
+* Coordinated Sampled Listening (CSL) Transmitter, for Full Thread Devices only.
```
Comment (greg-fer) [inline doc/nrf/ug_thread_configuring.rst:135]:
```suggestion
* Coordinated Sampled Listening (CSL) Transmitter (for Full Thread Devices only)
```

File: doc/nrf/ug_thread_configuring.rst [hunk 3]
```diff
@@ -143,6 +144,7 @@ Moreover, Thread 1.2 also comes with the following features that are supported f
 
 .. note::
    The Link Metrics and Coordinated Sampled Listening features are not supported for nRF53 Series devices yet.
+   The Backbone Router feature enables the Thread Network side functionality, but not the Backbone side functionality.
```
Comment (greg-fer) [inline doc/nrf/ug_thread_configuring.rst:147]:
```suggestion
   The Backbone Router feature enables the functionality for the Thread Network side, but not for the Backbone side.
```

File: doc/nrf/ug_thread_supported_features.rst [hunk 4]
```diff
@@ -35,6 +35,12 @@ In |NCS|, you can choose which version of the Thread protocol to use in your app
 By default, |NCS| supports Thread 1.1, but you can enable and configure Thread 1.2 by using :ref:`dedicated options <thread_ug_thread_1_2>`.
 
 .. note::
-    Not all Thread 1.2 functionalities are currently supported.
+    All Thread 1.2 mandatory functionalities are currently implemented, execept for full Border Router support.
```
Comment (greg-fer) [inline doc/nrf/ug_thread_supported_features.rst:38]:
```suggestion
    All Thread 1.2 mandatory functionalities are currently implemented, execept for the full Border Router support.
```

File: doc/nrf/ug_thread_supported_features.rst [hunk 5]
```diff
@@ -35,6 +35,12 @@ In |NCS|, you can choose which version of the Thread protocol to use in your app
...
     See :ref:`thread_ug_thread_1_2` for the list of 1.2 features that are currently available in |NCS|, with information about how to enable them.
     Currently, the :ref:`ot_cli_sample` sample is the only sample that provides an :ref:`ot_cli_sample_thread_v12`.
+
+    There are known limitations on the Thread 1.2 support:
```
Comment (greg-fer) [inline doc/nrf/ug_thread_supported_features.rst:42]: I would suggest a subsection:
```
Limitations for Thread 1.2 support
=====================

<content of lines 38-40>

The Thread 1.2 Specification support has the following limitations:

* The current implementation does not guarantee that all retransmitted frames will be secured when using the radio driver transmission security capabilities.
  For this reason , OpenThread retransmissions are disabled by default when the :option:`CONFIG_NRF_802154_ENCRYPTION` Kconfig option is enabled.
You can enable the retransmissions at your own risk.
* Due to code size limitation, the combination of complete set of Thread 1.2 features with the Bluetooth LE multiprotocol support is not possible for the nRF52833 DKs.
```
Are these limitations or known issues that should be tracked on the known issues page?

**Review by Damian-Nordic — APPROVED** [review]

**Review by greg-fer — CHANGES_REQUESTED** [review]

**Comment by MarekPorwisz** [comment]
I think we are also missing DUA here: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/releases/release-notes-latest.html#thread

File: doc/nrf/ug_thread_supported_features.rst (hunk 5 above)
Comment (edmont) [reply doc/nrf/ug_thread_supported_features.rst]: > Are these limitations or known issues that should be tracked on the known issues page?
I would say that's the case for the first one at least.

**Review by MarekPorwisz — APPROVED** [review]

**Review by hubertmis — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

File: doc/nrf/ug_thread_supported_features.rst [hunk 6]
```diff
@@ -35,6 +35,16 @@ In |NCS|, you can choose which version of the Thread protocol to use in your app
...
     See :ref:`thread_ug_thread_1_2` for the list of 1.2 features that are currently available in |NCS|, with information about how to enable them.
     Currently, the :ref:`ot_cli_sample` sample is the only sample that provides an :ref:`ot_cli_sample_thread_v12`.
+
+Limitations for Thread 1.2 support
+==================================
+
+The Thread 1.2 Specification support has the following limitations:
+
+* The current implementation does not guarantee that all retransmitted frames will be secured when using the radio driver transmission security capabilities.
+  For this reason , OpenThread retransmissions are disabled by default when the :option:`CONFIG_NRF_802154_ENCRYPTION` Kconfig option is enabled.
```
Comment (greg-fer) [inline doc/nrf/ug_thread_supported_features.rst:48]: Remove empty space before comma.

**Review by greg-fer — APPROVED** [review]
One space issue found.

**Comment by edmont** [comment]
Last minute missing addition to the release notes. CI was red because of unrelated CHIP test failure.

**Review by rlubos — APPROVED** [review]

**Pull request merged**